crates/common/mqtt_channel/src/config.rs

```diff
@@ -120,6 +128,24 @@ impl Config {
         }
     }
 
+    /// Set the last will message, this will be published when the mqtt connection gets closed.
+    pub fn with_last_will_message(
+        self,
+        topic: impl Into<String>,
+        payload: impl Into<Vec<u8>>,
+    ) -> Self {
+        let last_will_message = LastWill {
+            topic: topic.into(),
+            message: Bytes::from(payload.into()),
+            qos: rumqttc::QoS::AtLeastOnce,
+            retain: false,
```

Comment:
For the tedge daemons, retain must be set to true.

Reply:
Sure, now changed the signature of the `with_last_will_message` to give user more flexibility to pass the `qos` and `retain` flags.